src/api/trace/span.rs

```diff
@@ -38,6 +39,42 @@ pub trait Span: fmt::Debug + 'static + Send + Sync {
         self.add_event_with_timestamp(name, SystemTime::now(), attributes)
     }
 
+    /// Convenience method to record an exception/error as an `Event`
+    ///
+    /// An exception SHOULD be recorded as an Event on the span during which it occurred.
+    /// The name of the event MUST be "exception".
+    ///
+    /// The semantic conventions for Errors are described in ["Semantic Conventions for Exceptions"](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/exceptions.md)
+    ///
+    /// If user doesn't provide `error_type` or `stacktrace`, the corresponding attributes will not be set.
+    ///
+    /// User can custom exception message by overriding error's `fmt` method.
+    fn record_exception(
+        &self,
+        err: &dyn Error,
+        error_type: Option<String>,
```

Comment:
The error type name is often difficult for end users to produce, and the backtrace could eventually be pulled from  the err directly, maybe the initially we should just accept `err: &dyn Error,`? if the nightly feature does not stablize we could always add a `record_exception_with_backtrace` method later if people want the context.

Reply:
OK. Just thought add it since the spec provide this attribute. But it's kind of hard to get the type of error in runtime for Rust.